astropy/coordinates/tests/test_representation.py

```diff
@@ -1046,6 +1049,16 @@ class LogDRepresentation(BaseRepresentation):
                                         ('lat', Latitude),
                                         ('logr', u.Dex)])
 
+    # now try passing in a differential
+    diff = CartesianDifferential(d_x=1 * u.km/u.s,
```

Comment:
I think I correctly implemented `unit_vectors` and `scale_factors` but was just stuck on the dex issues. Just pushed the commits that contain my implementation: https://github.com/adrn/astropy/blob/469c02ef38902b7880638334f656248270989c26/astropy/coordinates/tests/test_representation.py#L1024